x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -0,0 +1,317 @@
+[role="xpack"]
+[[jwt-realm]]
+=== JWT authentication
+
+beta::[]
+
+{es} can be configured to trust JSON Web Tokens (JWTs) that are issued as an 
+authentication credential from an external service.
+
+When a JWT realm is used to authenticate with {es}, a distinction is made
+between the _client_ that is connecting to {es}, and the _user_ on whose behalf
+the request should run. The JWT identifies the user, and a separate credential
+is used to authenticate the client.
+
+A common scenario that uses JWTs is when an existing front-end application uses 
+OpenID Connect (OIDC) as an authentication method, and then accesses {es}
+on behalf of the authenticated user.
+
+TIP: If the front-end application does not exist, you can use the
+<<token-authentication-services>> instead.
+
+[[jwt-realm-oidc]]
+==== JWT uses OIDC workflows
+JWT authentication in {es} is derived from OIDC workflows, where users initiate
+contact with an OIDC Relying Party (RP) and are redirected to authenticate with
+an OIDC Provider (OP). Separate tokens are issued at different steps in the
+workflow to represent authentication, authorization, and identification.
+
+When users authenticate to the OP, they receive a _grant token_, which they
+present to the RP as proof of authentication. The RP then asks the OP to verify
+the  grant token and exchange it for two tokens: an _ID token_ and an
+_access token_.
+
+The ID token is a JWT, so the encoding is transparent; the other tokens are
+opaque. If the ID token is presented to a JWT realm, {es} can use it to
```

Comment:
It may be hard to get what "transparent" means here, and which other tokens we're talking about (given my previous suggestion to simplify and generalize the context). I don't think this is relevant here and it could be removed.

```suggestion
If the ID token is presented to a JWT realm, {es} can use it to
```

Reply:
I'll change this to read:
>If the ID token is presented to a JWT realm, {es} can use it to authenticate, identify, and authorize an individual user.